app/src/main/java/me/quadphase/qpdex/databaseAccess/PokemonFactory.java

```diff
@@ -1138,6 +1285,215 @@ private int getMoveID(Move move) {
+     */
+    public EggGroup[] getAllEggGroups() {
+
+        return eggGroups;
+    }
+
+    /**
+     * Getter to access all valid generations in PokemonFactory
+     *
+     * @return String[] array types of all valid generartions
+     */
+    public String[] getGenerations() {
+
+        return generations;
+    }
+    /**
+     * Database query, returns all uniqueIDs of pokemon with the abilityID in question
+     *
+     * @param abilityID of ability queried
+     * @return ArrayList of Integers containing uniqueIDs
+     */
+
+    public ArrayList<Integer> getAllUniqueIDsFromAbility(int abilityID){
+
+        int actualAbilityID = abilityID;
+        ArrayList<Integer> filteredUniqueids = new ArrayList<>();
+        ArrayList<Integer> filteredNationalids = new ArrayList<>();
+
+        if(actualAbilityID > -1) {
+            String[] selectionArg = {String.valueOf(actualAbilityID)};
+            Cursor cursor = database.query(POKEMON_ABILITIES_TABLE, null, ABILITIY_ID + "=?", selectionArg, null, null, null);
+            cursor.moveToFirst();
+            while (!cursor.isAfterLast()) {
+                int uniqueID = cursor.getInt(cursor.getColumnIndex(POKEMON_UNIQUE_ID));
+                filteredUniqueids.add(uniqueID);
+                cursor.moveToNext();
+            }
+            cursor.close();
+
+        }
+        return filteredUniqueids;
+
+    }
+
+    /**
+     * Filters all unique ID's that are of the type requested
+     *
+     * @param typeID of the type in question
+     * @return array list of ints corresponding to unique ids
+     */
+    public ArrayList<Integer> getAllUniqueIDsFromType(int typeID){
+
+        ArrayList<Integer> filteredUniqueids = new ArrayList<>();
+        ArrayList<Integer> filteredNationalids = new ArrayList<>();
+
+        if(typeID > 0) {
+            String[] selectionArg = {String.valueOf(typeID)};
+            Cursor cursor = database.query(POKEMON_TYPES_TABLE, null, TYPE_ID + "=?", selectionArg, null, null, null);
+            cursor.moveToFirst();
+            while (!cursor.isAfterLast()) {
+                int uniqueID = cursor.getInt(cursor.getColumnIndex(POKEMON_UNIQUE_ID));
+                filteredUniqueids.add(uniqueID);
+                cursor.moveToNext();
+            }
+            cursor.close();
+
+        }
+        return filteredUniqueids;
+
+    }
+
+    /**
+     * Filters all unique ID's that are of the bewteen the range of stats or the stat requested
+     * @param lowerLimit lower integer value limit
+     * @param upperLimit upper integer value
+     * @param stat the stat being filtered for, where stat is either: HP, ATTACK, DEFENCE, spattack, spdefence or SPEED
+     *             TODO stat can also be SUM, which will search for SUM
+     * @return array of integers of the unique id's that correspond
+     */
+    public ArrayList<Integer> getAllUniqueIDsFromStat(int lowerLimit, int upperLimit, String stat){
+        //TODO: Add validation step for stat string
```

Comment:
Reminder TODO!

Consider instead of a string, make it an enum with the possible values.